storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -550,12 +558,13 @@ public static void handleUncaughtException(Throwable t, Set<Class> allowedExcept
             }
         }
 
-        if(allowedExceptions.contains(t.getClass())) {
+        if(alwaysSwallow || allowedExceptions.contains(t.getClass())) {
```

Comment:
Even `alwaysSwallow` is set to true, we may still want to throw Error on `Error`s.